querybook/server/lib/query_analysis/validation/validators/presto_optimizing_validator.py

```diff
@@ -0,0 +1,260 @@
+from abc import ABCMeta, abstractmethod
+from typing import List, Tuple
+from sqlglot import TokenType
+from sqlglot.dialects import Trino
+from sqlglot.tokens import Token
+
+from lib.query_analysis.validation.base_query_validator import (
+    QueryValidationResult,
+    QueryValidationResultObjectType,
+    QueryValidationSeverity,
+)
+from lib.query_analysis.validation.validators.presto_explain_validator import (
+    PrestoExplainValidator,
+)
+
+UNION_ALL_SUGGESTION = "UNION ALL"
+APPROX_DISTINCT_SUGGESTION = "APPROX_DISTINCT("
+
+UNION_ALL_MESSAGE = "Using UNION ALL instead of UNION will execute faster"
+REGEXP_LIKE_MESSAGE = (
+    "Combining multiple LIKEs into one REGEXP_LIKE will execute faster"
+)
+APPROX_DISTINCT_MESSAGE = (
+    "Using APPROX_DISTINCT(x) instead of COUNT(DISTINCT x) will execute faster"
+)
+
+
+def _get_query_coordinate_by_index(query: str, index: int) -> Tuple[int, int]:
+    rows = query[: index + 1].splitlines(keepends=False)
+    return len(rows) - 1, len(rows[-1]) - 1
+
+
+def _tokenize_query(query: str) -> List[Token]:
+    return Trino.Tokenizer().tokenize(query)
```

Comment:
if these two util functions are only used by the validators here, I think we can also move them into the `BaseSQLGlotValidator` class